api/app/routers/land_change/dist_alerts/dist_alerts.py

```diff
@@ -0,0 +1,122 @@
+import json
+import uuid
+from pathlib import Path
+
+from fastapi import APIRouter, Request, BackgroundTasks, HTTPException
+from fastapi import Response as FastAPIResponse
+from fastapi.responses import ORJSONResponse
+
+from app.analysis.dist_alerts.analysis import do_analytics
+
+from app.models.common.analysis import AnalysisStatus
+from app.models.common.base import (
+    DataMartResourceLinkResponse,
+    DataMartResourceLink,
+)
+from app.models.land_change.dist_alerts import (
+    DistAlertsAnalyticsResponse,
+    DistAlertsAnalyticsIn,
+)
+
+
+router = APIRouter(prefix="/dist_alerts")
+
+PAYLOAD_STORE_DIR = Path("/tmp/dist_alerts_analytics_payloads")
+PAYLOAD_STORE_DIR.mkdir(parents=True, exist_ok=True)
+
+
+@router.post(
+    "/analytics",
+    response_class=ORJSONResponse,
+    response_model=DataMartResourceLinkResponse,
+    status_code=202,
+)
+def create(
+    *, data: DistAlertsAnalyticsIn, request: Request, background_tasks: BackgroundTasks
+):
+    # Convert model to JSON with sorted keys
+    payload_dict = data.model_dump()
+
+    # Convert to JSON string with sorted keys
+    payload_json = json.dumps(payload_dict, sort_keys=True)
+
+    # Generate deterministic UUID from payload
+    resource_id = str(uuid.uuid5(uuid.NAMESPACE_DNS, payload_json))
+
+    # Store payload in /tmp directory
+    payload_dir = PAYLOAD_STORE_DIR / resource_id
+    metadata_data = payload_dir / "metadata.json"
+    analytics_data = payload_dir / "data.json"
+
+    link = DataMartResourceLink(
+        link=f"{str(request.base_url).rstrip('/')}/v0/land_change/dist_alerts/analytics/{resource_id}"
+    )
+
+    if metadata_data.exists() and analytics_data.exists():
+        return DataMartResourceLinkResponse(data=link, status=AnalysisStatus.saved)
+
+    if metadata_data.exists():
+        return DataMartResourceLinkResponse(data=link, status=AnalysisStatus.pending)
+
+    payload_dir.mkdir(parents=True, exist_ok=True)
+    metadata_data.write_text(payload_json)
+    background_tasks.add_task(do_analytics, file_path=payload_dir)
+    return DataMartResourceLinkResponse(data=link, status=AnalysisStatus.pending)
+
+
+@router.get(
+    "/analytics/{resource_id}",
+    response_class=ORJSONResponse,
+    response_model=DistAlertsAnalyticsResponse,
+    status_code=200,
+)
+async def get_analytics_result(
+    resource_id: str,
+    response: FastAPIResponse,
+):
+    print("In the GET")
+    # Validate UUID format
+    try:
+        uuid.UUID(resource_id)
+    except ValueError:
+        raise HTTPException(
+            status_code=400, detail="Invalid resource ID format. Must be a valid UUID."
+        )
+
+    # Construct file path
+    file_path = PAYLOAD_STORE_DIR / resource_id
+    analytics_metadata = file_path / "metadata.json"
+    analytics_data = file_path / "data.json"
+    metadata_content = None
+    alerts_dict = None
+
+    if analytics_metadata.exists() and analytics_data.exists():
```

Comment:
I think here you need to still set metadata_content is just analytics_metadata.exists() is true so you can return a pending resource below

Reply:
oh, good catch! Yeah I think I see what you're talking about, @jterry64